icu4c/source/common/uresbund.cpp

```diff
@@ -1982,7 +2054,20 @@ ures_getByKeyWithFallback(const UResourceBundle *resB,
                     *status = U_USING_FALLBACK_WARNING;
                 }
 
-                fillIn = init_resb_result(dataEntry, res, inKey, -1, resB, fillIn, status);
+                fillIn = init_resb_result(dataEntry, res, key, -1, resB, fillIn, status);
+                if (resPath != nullptr) {
+                    createPath(path, resPath, len, origResPath, origResPathLen, inKey, status);
+                } else if (*(uprv_strchr(fillIn->fResPath, RES_PATH_SEPARATOR) + 1) != '\0') {
```

Comment:
fResPath can be nullptr (if the fillIn is for the rootRes of some bundle), so you should guard against that, unless there is earlier code in this function that guarantees it not to be nullptr.
And if the strchr() finds no `/` then it returns nullptr and the code crashes.

Reply:
Good catch.  Fixed.